Update aria-state-or-property-permitted-5c01ea.md (#1612)
act-rules/act-rules.github.io: _rules/aria-state-or-property-permitted-5c01ea.md

## Expectation

> One of the following is true for each test target:

Reviewer: Maybe just:

```suggestion
For each test target, one of the following is true:
```

- **Global**: the test target is a [global state or property][global]; or
- **Semantic Role**: the test target is an [inherited][], [supported][], or [required][] [state][] or [property][] of the [semantic role][] of the element on which the test target is specified; or

> - **language feature**: for an HTML element, the test target is allowed on the element on which it is specified. Which ARIA state or property may be used on which element is described in [ARIA in HTML](https://w3c.github.io/html-aria/).

Author: Lowercase it is.

## Expectation

> Each test target is either an [inherited][], [supported][], or [required][] [state][] or [property][] for the element's [semantic role][] or the [WAI-ARIA state or property][] may be used on the language feature. If the element has no [semantic role][], the attribute must be a [global state or property][global].

Reviewer 2: I'm a bit uneasy in leaving "language feature" as it and only explain it in a background note (versus trying to have a definition).
The rule is only targeting HTML and SVG, and I don't think there is a similar "ARIA in SVG" (or other feature document) out there. So maybe we can be a bit more prescriptive here (to make sure and remove the ambiguity). Pointing here to ARIA in HTML and saying that if the target is an HTML element, it may also have stuff described there.

Author: Yes, I thought about that too. In the end, I left the "language feature" so that everybody else can say what they think about it. It's unlikely, I think, there will be an SVG accessibility API mappings spec so we could tighten it.

Author: Yes, I thought about that too. In the end, I decided to leave the "language feature" for everybody else to decide on it. Yeah, I'm not sure if there ever will be an equivalent concept for SVG so we can tighten it.

Reviewer: I think you're trying to pack too much information into one sentence. It might be better to break this up into a list:

```suggestion
One of the following is true for each test target:

- **Global**: the test target is a [global state or property][global]; or

- **Semantic Role**: the test target is an [inherited][], [supported][], or [required][] [state][] or [property][] of the [semantic role][] of the element on which the test target is specified; or

- **language feature**: the test target is allowed as a language feature on the element on which it is specified. Which ARIA state or property may be used on which element is describe in [ARIA in HTML][].
```

This will also make it easier to add prohibited attributes when ARIA 1.2 makes it to rec.

## Background

> In HTML, there are language features that do not have corresponding implicit WAI-ARIA semantics. As per [ARIA in HTML Editor's Draft](https://w3c.github.io/html-aria/), those elements can have [global states or properties][global]. Some of those elements can also have [inherited][], [supported][], or [required][] [states][state] or [properties][property] that correspond to a [WAI-ARIA role](https://www.w3.org/TR/wai-aria-1.1/#usage_intro). For example, `audio` has no corresponding ARIA semantics but it can have [inherited][], [supported][], or [required][] [states][state] or [properties][property] of the [`application` role](https://www.w3.org/TR/wai-aria-1.1/#application).

Reviewer 2: ```suggestion
In HTML, there are language features that do not have corresponding implicit WAI-ARIA semantics. As per [ARIA in HTML Editor's Draft](https://w3c.github.io/html-aria/), those elements can have [global states or properties][global]. Some of those elements can also have [inherited][], [supported][], or [required][] [states][state] or [properties][property] that correspond to a [WAI-ARIA role](https://www.w3.org/TR/wai-aria-1.1/#usage_intro). For example, the `audio` element has no corresponding ARIA semantics but it can have [inherited][], [supported][], or [required][] [states][state] or [properties][property] of the [`application` role](https://www.w3.org/TR/wai-aria-1.1/#application).
```

Author: Done.

Assessing the value of the attribute is out of scope for this rule.

- [ARIA state or property has valid value](https://act-rules.github.io/rules/6a7281)
- [Understanding Success Criterion 4.1.1: Parsing](https://www.w3.org/WAI/WCAG21/Understanding/parsing.html)
- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
- [WAI-ARIA 1.1, Supported States and Properties](https://www.w3.org/TR/wai-aria-1.1/#states_and_properties)
- [WAI-ARIA 1.1, Global States and Properties](https://www.w3.org/TR/wai-aria-1.1/#global_states)
- [ARIA5: Using WAI-ARIA state and property attributes to expose the state of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA5)

> - [Document conformance requirements for use of ARIA attributes in HTML](https://w3c.github.io/html-aria/#docconformance)

Reviewer 2: All links to ARIA in HTML should be https://www.w3.org/TR/html-aria/ rather than github.io

Author: Done.

<button role="none" aria-pressed="false">ACT rules are cool!</button>
```

#### Passed Example 11

> This `input` element does not have an [explicit role][] of `textbox`, but the `aria-required` property may be used on such language feature.

Reviewer 2: ```suggestion
This `input` element does not have an [explicit role][] of `textbox`, but the `aria-required` property may be used `input` element with a `type` [attribute value][] of `password`.
```
+ probably linking to the precise section of ARIA in HTML that says so.
